substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/SubstrateGraphBuilderPlugins.java

```diff
@@ -371,6 +371,13 @@ private static void interceptProxyInterfaces(GraphBuilderContext b, ResolvedJava
              */
             List<Class<?>> classList = new ArrayList<>();
             FixedNode successor = unwrapNode(newArray.next());
+            if (successor instanceof EnsureClassInitializedNode) {
+                /*
+                 * Skip EnsureClassInitialized and Begin nodes.
+                 */
+                EnsureClassInitializedNode classInitializedNode = ((EnsureClassInitializedNode) successor);
+                successor = classInitializedNode.next().next();
```

Comment:
Instead of having a three-line comment, I would simply cast to a Begin node (potentially assert on it) and the do next. You say the same, but with code instead of the comment.

Reply:
Done.